compare.R

```diff
@@ -0,0 +1,81 @@
+library(ggplot2)
+
+# Compare multiple benchmark runs, generating graphs for each unique filesystem
+# Multiple runs for the same filesystem are averaged together
+
+save_plot <- function(plot, filename) {
+  ggsave(filename, plot=plot)
+  cat("Plot saved to ", filename)
+}
+
+args = commandArgs(trailingOnly = TRUE)
+
+if (length(args) == 0) {
+  msg <- "No arguments specified.
+Must specify one or more filesystem runs to display in the form <file system name>=<benchmark files glob>
+Example: Rscript compare.R ssd=*ssd.csv efs=*efs.csv"
+  stop(msg)
+}
+
+parsed_args <- regmatches(args, regexec("^(.+?)=(.+)$", args))
+bad_args <- which(vapply(parsed_args, length, integer(1)) == 0)
+if (length(bad_args) > 0) {
+  stop(sprintf("Invalid argument '%s' specified. Must be of the form <file system name>=<benchmark files glob>", args[[bad_args[1]]]))
+}
+
+arg_matrix <- do.call(rbind, parsed_args)
+files_to_read <- setNames(lapply(arg_matrix[,3], Sys.glob), arg_matrix[,2])
+bad_glob <- which(vapply(files_to_read, length, integer(1)) == 0)
+if (length(bad_glob) > 0) {
+  stop(sprintf("No files found for glob argument '%s'", files_to_read[[bad_glob[1]]]))
+}
+
+final_data_frame <- do.call(rbind, mapply(names(files_to_read), files_to_read, FUN = function(fs, files) {
+  # Read each file into a separate data frame, using read.csv(stringsAsFactors=FALSE)
+  data_frames <- lapply(files, read.csv, stringsAsFactors = FALSE)
+  # Combine all data frames into a single data frame
+  data_frame_all <- do.call(rbind, data_frames)
+  # Fix-up NAs for parallelism - set them to 1 as parallel tests never have a parallelization factor of 1
+  # This makes sure that all of the aggregation functions work correctly
+  data_frame_all$parallelism <- sapply(data_frame_all$parallelism, function(val){if (is.na(val)){ val <- 1};val})
```

Comment:
You can use `ifelse`, the vectorized version of if/else, here:
```suggestion
  data_frame_all$parallelism <- ifelse(is.na(data_frame_all$parallelism), 1, data_frame_all$parallelism)
```